### ISSUE TITLE
Clean up base path mapping in cloudformation resource

### ISSUE THREAD

**Issue by sheakelly** [dougmoscrop/serverless-plugin-custom-domain#6]
Would it be possible to implement the "DELETE" event type on the custom cloudformation resource so that it cleans up the base path mapping when the cfn stack is removed.

I am happy to do the work. Just wanted to get your feedback on the idea before I submit a PR

**Comment by dougmoscrop** [comment]
Sure I'm up for that, I initially avoided it because if you fail to delete it can make CF really unhappy (stack gets stuck in rollback) so I would suggest making it try and still "succeed" even if it fails after like a few retries?

**Comment by dougmoscrop** [comment]
Also a bunch of people have told me since I made this plugin the native CF support for base path mapping works predictably- have you gave that a shot? I just haven't got around to testing it. If it did I'd deprecate this plugin!

**Comment by sheakelly** [comment]
Ah ok. I did not realise that CF supported this. I will give it a go and let you know

**Comment by dougmoscrop** [comment]
Yeah at first it didn't, then it did, but it failed in weird ways, and now supposedly it's better - I appreciate you letting me know. We don't have any API gateways that have changed since I first wrote this plugin :)

**Comment by dls314** [comment]
I'm trying to figure out how to share management of the custom domain name and base path mappings between two versions of the serverless.yml for a v1/ and v2/ base path map. Being very accepting/safe seems to let this work today, but I think that implicitly deleting the custom domain name on stack deletion would break that (mis) use
